### ISSUE TITLE
uilist issues from migration #74341

### ISSUE THREAD

**Issue by Brambor** [CleverRaven/Cataclysm-DDA#75661]
### Describe the bug

Issues left in the code after merging #74341, see it for more info.

- [x] I cannot see loading individual components when loading a game. It just black-screens me for a while and then the game is loaded.
   - This shows for a frame or so
   - ![image](https://github.com/user-attachments/assets/36761ba8-af35-4bc0-8416-0bbbbb42c6dd)

- [x] The filter `/` is displayed behind the menu, as it is not yet written in ImGui.
   - ![image](https://github.com/user-attachments/assets/0f2d6417-f608-402a-b074-5fdbf46245fe)

- [ ] I would prefer on left mouse click, don't select the selected entry, but the entry under the cursor.
   - It's super annoying when clicking on the CDDA window to focus it and I select an entry instead. While testing a crash, this causes me to restart the game and reload the save. 
   - If the mouse is somewhere completely different, it feels weird:
   - ![image](https://github.com/user-attachments/assets/43fb7e87-661e-4d8d-8853-d14a74b4c485)

   - > I think it works correctly in all cases except when you click outside the menu. The mouse click is just translated to the SELECT command regardless of where the user actually clicked, and we need to fix that. Probably not in this patch, however. The problem is in an entirely different part of the code.

### Attach save file

N/A

### Steps to reproduce

issue 1: load a game
issues 2 & 3: go into the menu

### Expected behavior

1. See load screen.
2. The Interactible popup window is not behind the ImGui window.
3. Clicking works only if the entry is clicked.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.4780 (22H2)
- Game Version: 0.G-11377-g54be5797b6-dirty [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]

**Comment by GuardianDll** [comment]
someone also posted this in discord
![image](https://github.com/user-attachments/assets/b01cfaf1-1fc5-4389-92e8-bb7cf1f23792)

**Comment by PatrikLundell** [comment]
- I can confirm 1. That's really bad when you're forced to use debug builds, since those take an eternity to load.
- I can also confirm 2.
- I don't do mouse stuff, so 3 has to be done by someone else.
Since this is a multi issue report and I can't confirm all parts, I can't mark it as confirmed.

**Comment by HadeanLake** [comment]
I play with slightly increased font size
![fonts](https://github.com/user-attachments/assets/4c74e4c3-5a89-40ae-a996-33e1586823a4)


Window order is a problem - some windows are on top of others making it hard or impossible to read
MA selection is not only on top of list of techniques it also wider than screen, with MA names out of screen
![2024-08-14-141346_1920x1080_scrot](https://github.com/user-attachments/assets/af76c9b3-cf05-4d15-8f7e-820e0b1759ca)

Selecting components for crafting screen looks like this:
![crafting](https://github.com/user-attachments/assets/88b7189f-80b1-469a-aad1-fb1893813e0f)

When examining appliances list of actions obstructs information 
![power grid](https://github.com/user-attachments/assets/9c63c3a8-7ebf-4c60-890c-d33041f7175b)


![p2](https://github.com/user-attachments/assets/fccc2ec3-6bb6-4d48-822e-8c1411d7226c)

**Comment by Brambor** [comment]
> someone also posted this in discord ![image](https://private-user-images.githubusercontent.com/67688115/357833078-b01cfaf1-1fc5-4389-92e8-bb7cf1f23792.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM2NDM5OTMsIm5iZiI6MTcyMzY0MzY5MywicGF0aCI6Ii82NzY4ODExNS8zNTc4MzMwNzgtYjAxY2ZhZjEtMWZjNS00Mzg5LTkyZTgtYmI3Y2YxZjIzNzkyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE0VDEzNTQ1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlOWFhMTYzNjVmNmE2MTVjYjE2NzhiOTRjNWI3ZGMzN2FiNjA0NWZhN2NiNzg4ZGY1ZjIzMDQ5ZmFlOGYwYzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LrAVTluu4CU5q2F48W6LUvfK91ranCgq2JwgzEsaUZY)

Yes, this menu could use an improvement. The same goes for the debug menu:
![image](https://github.com/user-attachments/assets/824b627b-5fea-474f-8d1f-3de70380df99)

Note: broken on smaller screens:
![image](https://github.com/user-attachments/assets/69f7e52e-30a7-48db-8af9-1d2d2c3dae64)

**Comment by Brambor** [comment]
I might dust off the solution I proposed: get window size from uilist entries and line break all other text.

**Comment by Standing-Storm** [comment]
Mentioned elsewhere but putting here: the new menus do not process color tags, they just render them as text: 

![Untitled](https://github.com/user-attachments/assets/87e1b8df-4c4b-4472-809a-8dbed9e7b544)

**Comment by Terrorforge** [comment]
Spell menu also no longer shows the name of the current category. This is my Favorites section but it doesn't say so anywhere:
![image](https://github.com/user-attachments/assets/184638fd-5a4a-4c81-bee2-36e82989213b)

**Comment by db48x** [comment]
> Mentioned elsewhere but putting here: the new menus do not process color tags, they just render them as text:

Please be precise when reporting bugs! :D The new menus do clearly support color tags, it’s just that the description field of the Magiclysm spell casting menu ignores them.

The real problem is that our ImGui code doesn’t yet support coloring text if we are also wrapping it into a paragraph. See https://github.com/CleverRaven/Cataclysm-DDA/issues/75669 for more details.

**Comment by db48x** [comment]
> When examining appliances list of actions obstructs information 

Nice. The uilist can already be offset, so the caller in this specific case just needs to [calculate the correct offset](https://github.com/CleverRaven/Cataclysm-DDA/issues/75671).

**Comment by db48x** [comment]
> Selecting components for crafting screen looks like this: 

It looks like any menu with footer text or item descriptions has the same problem: color tags are shown to the user. I filed it separately as https://github.com/CleverRaven/Cataclysm-DDA/issues/75673

**Comment by db48x** [comment]
> Window order is a problem - some windows are on top of others making it hard or impossible to read MA selection is not only on top of list of techniques it also wider than screen, with MA names out of screen

I’ve filed the martial arts menu problem as https://github.com/CleverRaven/Cataclysm-DDA/issues/75674.

**Comment by db48x** [comment]
> I cannot see loading individual components when loading a game. It just black-screens me for a while and then the game is loaded.

Filed separately; https://github.com/CleverRaven/Cataclysm-DDA/issues/75678

>  The filter `/` is displayed behind the menu, as it is not yet written in ImGui.

Filed separately; https://github.com/CleverRaven/Cataclysm-DDA/issues/75676

>  I would prefer on left mouse click, don't select the selected entry, but the entry under the cursor.

Previously filed as https://github.com/CleverRaven/Cataclysm-DDA/issues/75288

**Comment by Brambor** [comment]
> > I might dust off the solution I proposed: get window size from uilist entries and line break all other text.
> 
> Try it and see how it looks. If it looks better I’ll take it. However, in this case I bet it makes the window really skinny and wraps the text to unreadability.

You might be right. Adding min max height to the caller might fix it, if my solution doesn't work.

I also considered a hard max to the CDDA window size, but entries cannot be broken, so it would affect only the title and description. Wrapping them to the whole width of the window would solve unreadability, but it would still look horrendous.

**Comment by akrieger** [comment]
Pocket settings menu has some weird overlaps, and the submenu renders behind everything else.
<img width="1244" alt="image" src="https://github.com/user-attachments/assets/97ca010f-4a4f-4193-93d0-e661b90e1c08">

**Comment by akrieger** [comment]
It's like it's not completely clearing the old screen first or something. There's an afterimage until it renders another frame.
<img width="592" alt="image" src="https://github.com/user-attachments/assets/f1cbf746-6f51-48b6-aafc-95e0e9847fbe">

**Comment by akrieger** [comment]
Pockets settings left list doesn't scroll when the cursor wraps around from bottom to top (or vice versa) until another scroll event happens.

**Comment by akrieger** [comment]
Some cursed triple selection bars if you reverse directions int he middle of the list. 
<img width="410" alt="image" src="https://github.com/user-attachments/assets/329e2df1-561d-4692-9f31-c4e6a5a17af7">

**Comment by db48x** [comment]
> Pocket settings menu has some weird overlaps, …

What is overlapping in that screenshot? The blue border overlaps with the text, but that’s already a known problem.

**Comment by akrieger** [comment]
That was it, the blue box.

**Comment by failboat78** [comment]
Can't read bottom of firearm mend menu 
![image](https://github.com/user-attachments/assets/aa65f5e6-acca-4e21-b9da-e3057f691382)

**Comment by db48x** [comment]
Thanks, those are both great examples of #75698.

**Comment by Brambor** [comment]
Everything here is addressed or filed separately. Except for this:

> Pocket settings menu has some weird overlaps, and the submenu renders behind everything else. <img alt="image" width="1244" src="https://private-user-images.githubusercontent.com/667719/358118951-97ca010f-4a4f-4193-93d0-e661b90e1c08.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3ODQxMjksIm5iZiI6MTczMzc4MzgyOSwicGF0aCI6Ii82Njc3MTkvMzU4MTE4OTUxLTk3Y2EwMTBmLTRhNGYtNDE5My05M2QwLWU2NjFiOTBlMWMwOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwOVQyMjM3MDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNzUxNDQ4M2MxNDgzZjM0NDUyZjhkZDE5YTQzMjQ1ODc3NTlkOTlhOWUyMDQ4ZTliMzkzNzhjMmQzZTZjYmIzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.2E6TaUD3scb5Azf1mPp80WU2C3i4IsgSGdPrjFaQlkI">

The priority menu is rendered behind the ImGui menu. I will fix this now.

**Comment by Brambor** [comment]
> Some cursed triple selection bars if you reverse directions in the middle of the list when using arrow keys. <img alt="image" width="410" src="https://private-user-images.githubusercontent.com/667719/358120784-329e2df1-561d-4692-9f31-c4e6a5a17af7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3ODc5MDMsIm5iZiI6MTczMzc4NzYwMywicGF0aCI6Ii82Njc3MTkvMzU4MTIwNzg0LTMyOWUyZGYxLTU2MWQtNDY5Mi05ZjMxLWM0ZTZhNWExN2FmNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwOVQyMzQwMDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MGFhYzY3MGM5MThiOGQ1OTFiYTg5MTYzZDc3YzljNGViNWM4MDIxOGRlMGJjZTg2Nzk0YjY2YmJlYTJjN2EzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.I4f7j7hxwGxbjEHT6KmqOpwK-hnT9Go_4ek8X9X-29Q">

I did reproduce that before, somehow, but not after #78453

**Comment by Brambor** [comment]
> Pockets settings left list doesn't scroll when the cursor wraps around from bottom to top (or vice versa) until another scroll event happens.

I can reproduce this only with PageDown not wrapping to the top. At all.

Home jumps to the last entry, as does End.

But these are for a new issue, this is too cluttered already.
